mipdb/dataset.py

```diff
@@ -26,5 +32,74 @@ def _verify_dataset_field(self):
         if len(set(self.data["dataset"])) > 1:
             raise InvalidDatasetError("The dataset field contains multiple values.")
 
+    def validate_dataset(self, metadata_table):
+        pa_columns = {}
+        # Validating that the dataset always has subjectcode as the first column.
+        # This is according to the data requirements
+        # https://github.com/HBPMedical/mip-deployment/blob/master/documentation/NewDataRequirements.md
+        if "subjectcode" != self.data.keys()[0]:
+            raise InvalidDatasetError(
+                f"""Error inserting dataset into the database,
+                    subjectcode should be the first column.
+                """
+            )
+
+        columns = [column for column in self.data if column != "subjectcode"]
+        # There is a need to construct a DataFrameSchema with all the constrains that the metadata is imposing
+        # For each column a pandera Column is created that will contain the constrains for the specific column
+        for column in columns:
+            if column not in metadata_table:
+                raise InvalidDatasetError(f"The column: '{column}' does not exist in the metadata")
+            metadata_column = metadata_table[column].metadata
+            metadata_column_dict = json.loads(metadata_column)
+            checks = self._get_pa_checks(metadata_column_dict)
+            pa_type = self.pa_type_from_sql_type(metadata_column_dict["sql_type"])
+            pa_columns[column] = pa.Column(dtype=pa_type, checks=checks, nullable=True)
+
+        index = pa.Index(
+            pa.String,
+            checks=[
+                # id is unique
+                pa.Check(lambda s: s.duplicated().sum() == 0),
+            ],
+        )
```

Comment:
This index constraint just verifies that the default auto-incremental index of the Dataframe has unique values. In order to make pandera check the "subjectcode" column you have to say to pandas that it should be the index column in `__init__`:
`self._data = self._data.set_index('subjectcode')`